state-sync/src/network.rs

```diff
@@ -57,13 +58,9 @@ impl NewNetworkSender for StateSyncSender {
 }
 
 impl StateSyncSender {
-    pub fn send_to(
-        &mut self,
-        recipient: PeerId,
-        message: StateSyncMessage,
-    ) -> Result<(), NetworkError> {
+    pub fn send_to(&mut self, recipient: PeerId, message: StateSyncMessage) -> Result<(), Error> {
         let protocol = ProtocolId::StateSyncDirectSend;
-        self.inner.send_to(recipient, protocol, message)
+        Ok(self.inner.send_to(recipient, protocol, message)?)
```

Comment:
Ugh, I will never understand this still how the `Ok(result?)` will upgrade the send_to error to an `Error` but you can't just put the result without doing the `Ok(?)`

Reply:
Hah 😄  We just need to convert the error to the right type. We could also do `result.map_err(|error| error.into)` but it's longer. See: https://doc.rust-lang.org/edition-guide/rust-2018/error-handling-and-panics/the-question-mark-operator-for-easier-error-handling.html

Reply:
Ah, I didn't realize it did the `into()`